ucsmsdk/ucsdriver.py

```diff
@@ -84,7 +84,7 @@ def connect(self):
             ssl_context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
             ssl_context.options |= ssl.OP_NO_SSLv2
             ssl_context.options |= ssl.OP_NO_SSLv3
-            if self.key_file and self.cert_file:
+            if getattr(self, "key_file", None) and getattr(self, "cert_file", None):
```

Comment:
Can you provide context in PR description or Code comments around when key_file and/or cert_file won't be present?

Reply:
Before getattr() implement case: We are just checked the value of cert_file and key_file is present or not. i.e, it is throwing error with python3.12 and works with python3.11.
AttributeError: 'TLSConnection' object has no attribute 'key_file' 

After we used getattr () -> to get the value of an object's attribute and if no attribute of that object is found, default value is returned, so error resolved works on both python versions.

Reply:
If key_file or cert_file is not defined or their values are None, the getattr() function will return None for that attribute. So, if key_file or cert_file is not set or set to None the condition in the if statement will evaluate to False, and the subsequent code block will not execute.
In our case both are defined are None so, the code block is not skipped.
So, we did fix to avoid any AttributeError in both conditions.